core/metacling/src/TCling.cxx

```diff
@@ -8928,6 +8933,8 @@ void TCling::FuncTempInfo_Title(FuncTempInfo_t *ft_info, TString &output) const
 
 void TCling::MethodInfo_Delete(MethodInfo_t* minfo) const
 {
+   // Object deletion should lock, do it here instead of at the call site.
```

Comment:
Why?  As far as I can the destruction of `TClingMethodInfo` is passive and does not need the lock.